packages/fast-jest-snapshots-react/src/index.ts

```diff
@@ -55,12 +67,23 @@ export function renderSnapshot<T>(data: T, component: React.ComponentClass<T>, t
  * Generate a set of snapshot tests given a snapshot suite
  */
 export function generateSnapshots<T>(examples: ISnapshotTestSuite<T>): void {
-    const data: T[] = examples.data;
+    const data: ISnapshotTestCases<T> = examples.data;
     const component: React.ComponentClass<T> = examples.component;
 
     if (Array.isArray(data)) {
-        data.forEach((example: T, index: number): void => {
-            renderSnapshot(example, component, `${examples.name}: ${index}`);
+        data.forEach((example: ISnapshotTestCase<T>, index: number): void => {
+            let title: string;
+            let props: T;
+
+            if (Array.isArray(example)) {
+                title = `${examples.name} ${example[0]}`;
+                props = example[1];
+            } else {
+                title = `${examples.name}: ${index}`;
+                props = example;
+            }
+
+            renderSnapshot(props, component, title);

```

Comment:
could this be simplified, eg:
```
const isArray: boolean = Array.isArray(example);

renderSnapshot(
    isArray ? `${examples.name} ${example[0]}` : `${examples.name}: ${index}`
    component,
    isArray ? example[1] : example
);
```

Reply:
I'm going to leave as-is. IMO this is more readable